## 71% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 2
Expected to fail tests: 0
Passed tests: 5
```
```
IDs of failed tests: 727230,727229 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=feature/Backup_profile_recovery_%2321366&PR_ID=21408&APK_NAME=21408.apk&TR_CASE_IDS=727230,727229">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693908">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td>

```
Test setup failed: critical/test_wallet.py:29: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:52: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:297: in recover_access
    self.terms_and_privacy_checkbox.click()
../views/base_element.py:92: in click
    element.click()
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:94: in click
    self._execute(Command.CLICK_ELEMENT)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 The element 'By.xpath: //*[@content-desc='terms-privacy-checkbox-container']/*[@content-desc='checkbox-off']' does not exist in DOM anymore; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
E   Stacktrace:
E   io.appium.uiautomator2.common.exceptions.StaleElementReferenceException: The element 'By.xpath: //*[@content-desc='terms-privacy-checkbox-container']/*[@content-desc='checkbox-off']' does not exist in DOM anymore
E   	at io.appium.uiautomator2.model.ElementsCache.restore(ElementsCache.java:117)
E   	at io.appium.uiautomator2.model.ElementsCache.get(ElementsCache.java:153)
E   	at io.appium.uiautomator2.handler.Click.safeHandle(Click.java:36)
E   	at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:59)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:259)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:253)
E   	at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:77)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
E   	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
E   	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
E   	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
E   	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:611)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:552)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:466)
E   	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438)
E   	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
E   	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
E   	at java.lang.Thread.run(Thread.java:1012)
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693907">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>## Multiaccount is recovered successfully!</div></blockquote></p>

```
Test setup failed: critical/test_wallet.py:29: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:52: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/sign_in_view.py:297: in recover_access
    self.terms_and_privacy_checkbox.click()
../views/base_element.py:92: in click
    element.click()
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:94: in click
    self._execute(Command.CLICK_ELEMENT)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 The element 'By.xpath: //*[@content-desc='terms-privacy-checkbox-container']/*[@content-desc='checkbox-off']' does not exist in DOM anymore; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
E   Stacktrace:
E   io.appium.uiautomator2.common.exceptions.StaleElementReferenceException: The element 'By.xpath: //*[@content-desc='terms-privacy-checkbox-container']/*[@content-desc='checkbox-off']' does not exist in DOM anymore
E   	at io.appium.uiautomator2.model.ElementsCache.restore(ElementsCache.java:117)
E   	at io.appium.uiautomator2.model.ElementsCache.get(ElementsCache.java:153)
E   	at io.appium.uiautomator2.handler.Click.safeHandle(Click.java:36)
E   	at io.appium.uiautomator2.handler.request.SafeRequestHandler.handle(SafeRequestHandler.java:59)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleRequest(AppiumServlet.java:259)
E   	at io.appium.uiautomator2.server.AppiumServlet.handleHttpRequest(AppiumServlet.java:253)
E   	at io.appium.uiautomator2.http.ServerHandler.channelRead(ServerHandler.java:77)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
E   	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
E   	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
E   	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:266)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:345)
E   	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1294)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:366)
E   	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:352)
E   	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
E   	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:611)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:552)
E   	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:466)
E   	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:438)
E   	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
E   	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
E   	at java.lang.Thread.run(Thread.java:1012)
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693903">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f1f74aa650bb4a559b0b303a5df71221?auth=4ab626e4a08547baf4404e5d54eda90b#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693906">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f1f74aa650bb4a559b0b303a5df71221?auth=4ab626e4a08547baf4404e5d54eda90b#95">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693909">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b18fd3f33d5f48539e8e309359d6d15c?auth=cfa2c3c45a4ceddf7d4429b1580d05e9#42">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693904">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9033732f55d84ee69970ab1a814d1380?auth=da64b6069016ae1a2c70c1f05ba23497#100">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/59327ae86bdf4a17a6cf79a9ab573c16?auth=7793c2788e9241cb858e743bc1b0d9a2#91">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3693905">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/dd72e7b9568b4adfb1774807949efadb?auth=72f1fdb0c2926b414bc3cc7081298e8e#168">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4e1a8ac6d8154c72aef252436db2838e?auth=f15a96c566c6a6a0c4f636409a20c87e#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/common/enter_seed_phrase/view.cljs
+++ b/src/status_im/common/enter_seed_phrase/view.cljs
@@ -96,7 +96,7 @@
        (take 7)))
 
 (defn recovery-phrase-screen
-  [{:keys [banner-offset initial-insets keypair title recovering-keypair? render-controls]}]
+  [{:keys [banner-offset initial-insets keypair title on-success render-controls]}]
   (reagent/with-let [keyboard-shown?           (reagent/atom false)
                      keyboard-show-listener    (.addListener rn/keyboard
                                                              "keyboardDidShow"
@@ -120,14 +120,20 @@
                                                  (reset! seed-phrase new-phrase))
                      on-submit                 (fn []
                                                  (swap! seed-phrase clean-seed-phrase)
-                                                 (if recovering-keypair?
-                                                   (rf/dispatch [:wallet/seed-phrase-entered
-                                                                 (security/mask-data
-                                                                  @seed-phrase)
-                                                                 set-invalid-seed-phrase])
-                                                   (rf/dispatch [:onboarding/seed-phrase-entered
-                                                                 (security/mask-data @seed-phrase)
-                                                                 set-invalid-seed-phrase])))]
+                                                 (rf/dispatch
+                                                  [:profile.recover/validate-recovery-phrase
+                                                   (security/mask-data @seed-phrase)
+                                                   {:on-success (fn [mnemonic key-uid]
+                                                                  (if on-success
+                                                                    (on-success
+                                                                     {:key-uid key-uid
+                                                                      :phrase mnemonic
+                                                                      :on-error
+                                                                      set-invalid-seed-phrase})
+                                                                    (rf/dispatch
+                                                                     [:onboarding/seed-phrase-validated
+                                                                      mnemonic key-uid])))
+                                                    :on-error   set-invalid-seed-phrase}]))]
     (let [words-coll               (mnemonic/passphrase->words @seed-phrase)
           last-word                (peek words-coll)
           pick-suggested-word      (fn [pressed-word]
@@ -191,7 +197,7 @@
          [rn/view {:style style/keyboard-container}
           [quo/predictive-keyboard
            {:type     suggestions-state
-            :blur?    (not recovering-keypair?)
+            :blur?    (not on-success)
             :text     suggestions-text
             :words    (keyboard-suggestions last-word)
             :on-press pick-suggested-word}]])])
@@ -200,7 +206,7 @@
      (.remove keyboard-hide-listener))))
 
 (defn screen
-  [{:keys [title keypair navigation-icon recovering-keypair? render-controls]}]
+  [{:keys [title keypair navigation-icon on-success render-controls]}]
   (let [[insets _]    (rn/use-state (safe-area/get-insets))
         banner-offset (rf/sub [:alert-banners/top-margin])]
     [rn/view {:style style/full-layout}
@@ -209,21 +215,22 @@
        {:margin-top (:top insets)
         :background :blur
         :icon-name  (or navigation-icon
-                        (if recovering-keypair? :i/close :i/arrow-left))
+                        (if on-success :i/close :i/arrow-left))
         :on-press   #(rf/dispatch [:navigate-back])}]
       [recovery-phrase-screen
-       {:title               title
-        :keypair             keypair
-        :render-controls     render-controls
-        :banner-offset       banner-offset
-        :initial-insets      insets
-        :recovering-keypair? recovering-keypair?}]]]))
+       {:title           title
+        :keypair         keypair
+        :render-controls render-controls
+        :banner-offset   banner-offset
+        :initial-insets  insets
+        :on-success      on-success}]]]))
 
 (defn view
   []
-  (let [{:keys [recovering-keypair?]} (rf/sub [:get-screen-params])]
+  (let [{:keys [on-success]} (rf/sub [:get-screen-params])]
     (rn/use-unmount
-     #(rf/dispatch [:onboarding/clear-navigated-to-enter-seed-phrase-from-screen]))
+     (when-not on-success
+       #(rf/dispatch [:onboarding/clear-navigated-to-enter-seed-phrase-from-screen])))
     [screen
-     {:title               (i18n/label :t/use-recovery-phrase)
-      :recovering-keypair? recovering-keypair?}]))
+     {:title      (i18n/label :t/use-recovery-phrase)
+      :on-success on-success}]))

--- a/src/status_im/contexts/keycard/events.cljs
+++ b/src/status_im/contexts/keycard/events.cljs
@@ -72,22 +72,21 @@
 
 (rf/reg-event-fx :keycard/on-get-application-info-success
  (fn [{:keys [db]} [application-info {:keys [key-uid on-success-fx]}]]
-   (let [error (keycard.utils/validate-application-info key-uid application-info)]
-     (if error
-       (case error
-         :keycard/error.not-keycard
-         {:fx [[:dispatch [:keycard/disconnect]]
-               [:dispatch [:open-modal :screen/keycard.not-keycard]]]}
-         :keycard/error.keycard-blank
-         {:fx [[:dispatch [:keycard/disconnect]]
-               [:dispatch [:open-modal :screen/keycard.empty]]]}
-         {:db (assoc-in db [:keycard :application-info-error] error)
-          :fx [[:dispatch [:keycard/disconnect]]
-               [:dispatch [:open-modal :screen/keycard.error]]]})
-       {:db (-> db
-                (assoc-in [:keycard :application-info] application-info)
-                (assoc-in [:keycard :pin :status] :verifying))
-        :fx on-success-fx}))))
+   (if-let [error (keycard.utils/validate-application-info key-uid application-info)]
+     (case error
+       :keycard/error.not-keycard
+       {:fx [[:dispatch [:keycard/disconnect]]
+             [:dispatch [:open-modal :screen/keycard.not-keycard]]]}
+       :keycard/error.keycard-blank
+       {:fx [[:dispatch [:keycard/disconnect]]
+             [:dispatch [:open-modal :screen/keycard.empty]]]}
+       {:db (assoc-in db [:keycard :application-info-error] error)
+        :fx [[:dispatch [:keycard/disconnect]]
+             [:dispatch [:open-modal :screen/keycard.error]]]})
+     {:db (-> db
+              (assoc-in [:keycard :application-info] application-info)
+              (assoc-in [:keycard :pin :status] :verifying))
+      :fx on-success-fx})))
 
 (rf/reg-event-fx :keycard/get-application-info
  (fn [_ [{:keys [on-success on-failure]}]]

--- a/src/status_im/contexts/keycard/sheets/migrate/view.cljs
+++ b/src/status_im/contexts/keycard/sheets/migrate/view.cljs
@@ -6,9 +6,10 @@
 
 (defn view
   []
-  (let [profile-name        (rf/sub [:profile/name])
-        profile-picture     (rf/sub [:profile/image])
-        customization-color (rf/sub [:profile/customization-color])]
+  (let [profile-name               (rf/sub [:profile/name])
+        profile-picture            (rf/sub [:profile/image])
+        customization-color        (rf/sub [:profile/customization-color])
+        recovery-phrase-backed-up? (rf/sub [:profile/recovery-phrase-backed-up?])]
     [:<>
      [quo/drawer-top
       {:type                :context-tag
@@ -27,7 +28,13 @@
      [quo/bottom-actions
       {:actions          :two-actions
        :button-one-label (i18n/label :t/continue)
-       :button-one-props {:on-press #()}
+       :button-one-props {:on-press #(if recovery-phrase-backed-up?
+                                       (rf/dispatch [:open-modal :screen/use-recovery-phrase
+                                                     {:on-success (fn [{:keys [key-uid phrase
+                                                                               on-error]}])}])
+                                       (rf/dispatch [:open-modal :screen/backup-recovery-phrase
+                                                     {:on-success (fn [{:keys [masked-seed-phrase
+                                                                               random-phrase]}])}]))}
        :button-two-label (i18n/label :t/cancel)
        :button-two-props {:type     :grey
                           :on-press #(rf/dispatch [:hide-bottom-sheet])}}]]))

--- a/src/status_im/contexts/keycard/sheets/migrate/view.cljs
+++ b/src/status_im/contexts/keycard/sheets/migrate/view.cljs
@@ -30,11 +30,9 @@
        :button-one-label (i18n/label :t/continue)
        :button-one-props {:on-press #(if recovery-phrase-backed-up?
                                        (rf/dispatch [:open-modal :screen/use-recovery-phrase
-                                                     {:on-success (fn [{:keys [key-uid phrase
-                                                                               on-error]}])}])
+                                                     {:on-success (fn [])}])
                                        (rf/dispatch [:open-modal :screen/backup-recovery-phrase
-                                                     {:on-success (fn [{:keys [masked-seed-phrase
-                                                                               random-phrase]}])}]))}
+                                                     {:on-success (fn [])}]))}
        :button-two-label (i18n/label :t/cancel)
        :button-two-props {:type     :grey
                           :on-press #(rf/dispatch [:hide-bottom-sheet])}}]]))

--- a/src/status_im/contexts/onboarding/events.cljs
+++ b/src/status_im/contexts/onboarding/events.cljs
@@ -131,15 +131,6 @@
                  [:open-modal :screen/onboarding.enable-biometrics]
                  [:open-modal :screen/onboarding.enable-notifications])}))
 
-(rf/defn seed-phrase-entered
-  {:events [:onboarding/seed-phrase-entered]}
-  [_ seed-phrase on-error]
-  {:multiaccount/validate-mnemonic [seed-phrase
-                                    (fn [mnemonic key-uid]
-                                      (re-frame/dispatch [:onboarding/seed-phrase-validated
-                                                          mnemonic key-uid]))
-                                    on-error]})
-
 (rf/defn seed-phrase-validated
   {:events [:onboarding/seed-phrase-validated]}
   [{:keys [db]} seed-phrase key-uid]

--- a/src/status_im/contexts/profile/backup_recovery_phrase/style.cljs
+++ b/src/status_im/contexts/profile/backup_recovery_phrase/style.cljs
@@ -1,4 +1,4 @@
-(ns status-im.contexts.wallet.add-account.create-account.new-keypair.backup-recovery-phrase.style
+(ns status-im.contexts.profile.backup-recovery-phrase.style
   (:require
     [quo.foundations.colors :as colors]
     [react-native.platform :as platform]))

--- a/src/status_im/contexts/profile/backup_recovery_phrase/view.cljs
+++ b/src/status_im/contexts/profile/backup_recovery_phrase/view.cljs
@@ -1,4 +1,4 @@
-(ns status-im.contexts.wallet.add-account.create-account.new-keypair.backup-recovery-phrase.view
+(ns status-im.contexts.profile.backup-recovery-phrase.view
   (:require
     [clojure.string :as string]
     [native-module.core :as native-module]
@@ -6,8 +6,7 @@
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
     [reagent.core :as reagent]
-    [status-im.contexts.wallet.add-account.create-account.new-keypair.backup-recovery-phrase.style :as
-     style]
+    [status-im.contexts.profile.backup-recovery-phrase.style :as style]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
@@ -45,20 +44,20 @@
 
 (defn view
   []
-  (let [step-labels         [:t/backup-step-1 :t/backup-step-2 :t/backup-step-3
-                             :t/backup-step-4]
-        checked?            (reagent/atom
-                             {:0 false
-                              :1 false
-                              :2 false
-                              :3 false})
-        revealed?           (reagent/atom false)
-        customization-color (rf/sub [:profile/customization-color])
-        seed-phrase         (reagent/atom [])
-        random-phrase       (reagent/atom [])]
+  (let [step-labels          [:t/backup-step-1 :t/backup-step-2 :t/backup-step-3
+                              :t/backup-step-4]
+        checked?             (reagent/atom
+                              {:0 false
+                               :1 false
+                               :2 false
+                               :3 false})
+        revealed?            (reagent/atom false)
+        customization-color  (rf/sub [:profile/customization-color])
+        {:keys [on-success]} (rf/sub [:get-screen-params])
+        seed-phrase          (reagent/atom [])
+        random-phrase        (reagent/atom [])]
     (fn []
       (let [theme (quo.theme/use-theme)]
-
         (rn/use-mount
          (fn []
            (native-module/get-random-mnemonic #(reset! seed-phrase (string/split % #"\s")))
@@ -107,10 +106,10 @@
               :button-one-label (i18n/label :t/i-have-written)
               :button-one-props {:disabled?           (some false? (vals @checked?))
                                  :customization-color customization-color
-                                 :on-press            #(rf/dispatch [:wallet/store-new-seed-phrase
-                                                                     {:seed-phrase   (security/mask-data
-                                                                                      @seed-phrase)
-                                                                      :random-phrase @random-phrase}])}}]
+                                 :on-press            #(on-success {:masked-seed-phrase
+                                                                    (security/mask-data
+                                                                     @seed-phrase)
+                                                                    :random-phrase @random-phrase})}}]
             [quo/text
              {:size  :paragraph-2
               :style (style/description-text theme)}

--- a/src/status_im/contexts/profile/recover/effects.cljs
+++ b/src/status_im/contexts/profile/recover/effects.cljs
@@ -1,9 +1,31 @@
 (ns status-im.contexts.profile.recover.effects
   (:require
     [native-module.core :as native-module]
-    [utils.re-frame :as rf]))
+    [promesa.core :as promesa]
+    [utils.re-frame :as rf]
+    [utils.security.core :as security]
+    [utils.transforms :as transforms]))
 
 (rf/reg-fx :effects.profile/restore-and-login
  (fn [request]
    ;;"node.login" signal will be triggered as a callback
    (native-module/restore-account-and-login request)))
+
+(defn validate-mnemonic
+  [mnemonic]
+  (-> mnemonic
+      (security/safe-unmask-data)
+      (native-module/validate-mnemonic)
+      (promesa/then (fn [result]
+                      (let [{:keys [keyUID]} (transforms/json->clj result)]
+                        {:key-uid keyUID})))))
+
+(rf/reg-fx :effects.profile/validate-recovery-phrase
+ (fn [[mnemonic on-success on-error]]
+   (-> (validate-mnemonic mnemonic)
+       (promesa/then (fn [{:keys [key-uid]}]
+                       (when (fn? on-success)
+                         (on-success mnemonic key-uid))))
+       (promesa/catch (fn [error]
+                        (when (and error (fn? on-error))
+                          (on-error error)))))))

--- a/src/status_im/contexts/profile/recover/events.cljs
+++ b/src/status_im/contexts/profile/recover/events.cljs
@@ -24,3 +24,7 @@
             :imagePath          (profile.config/strip-file-prefix image-path)
             :customizationColor (or color constants/profile-default-color)
             :fetchBackup        true)}))
+
+(rf/reg-event-fx :profile.recover/validate-recovery-phrase
+ (fn [_ [phrase {:keys [on-success on-error]}]]
+   {:effects.profile/validate-recovery-phrase [phrase on-success on-error]}))

--- a/src/status_im/contexts/profile/settings/view.cljs
+++ b/src/status_im/contexts/profile/settings/view.cljs
@@ -49,13 +49,14 @@
 
 (defn view
   []
-  (let [theme               (quo.theme/use-theme)
-        insets              (safe-area/get-insets)
-        customization-color (rf/sub [:profile/customization-color])
-        scroll-y            (reanimated/use-shared-value 0)
-        profile             (rf/sub [:profile/profile])
-        full-name           (profile.utils/displayed-name profile)
-        on-scroll           (rn/use-callback #(scroll-handler % scroll-y))]
+  (let [theme                      (quo.theme/use-theme)
+        insets                     (safe-area/get-insets)
+        customization-color        (rf/sub [:profile/customization-color])
+        scroll-y                   (reanimated/use-shared-value 0)
+        profile                    (rf/sub [:profile/profile])
+        recovery-phrase-backed-up? (rf/sub [:profile/recovery-phrase-backed-up?])
+        full-name                  (profile.utils/displayed-name profile)
+        on-scroll                  (rn/use-callback #(scroll-handler % scroll-y))]
     [quo/overlay {:type :shell}
      [rn/view
       {:style (style/navigation-wrapper {:customization-color customization-color
@@ -79,7 +80,7 @@
                                                                      profile)}}])}]}]]
      [rn/flat-list
       {:header                          [settings.header/view {:scroll-y scroll-y}]
-       :data                            (settings.items/items (boolean (seq (:mnemonic profile))))
+       :data                            (settings.items/items (not recovery-phrase-backed-up?))
        :shows-vertical-scroll-indicator false
        :render-fn                       settings-category-view
        :get-item-layout                 get-item-layout

--- a/src/status_im/contexts/settings/wallet/events.cljs
+++ b/src/status_im/contexts/settings/wallet/events.cljs
@@ -160,7 +160,7 @@
 
 (defn wallet-validate-seed-phrase
   [_ [seed-phrase on-success on-error]]
-  {:fx [[:multiaccount/validate-mnemonic [seed-phrase on-success on-error]]]})
+  {:fx [[:effects.profile/validate-recovery-phrase [seed-phrase on-success on-error]]]})
 
 (rf/reg-event-fx :wallet/validate-seed-phrase wallet-validate-seed-phrase)
 

--- a/src/status_im/contexts/settings/wallet/events_test.cljs
+++ b/src/status_im/contexts/settings/wallet/events_test.cljs
@@ -101,7 +101,7 @@
         seed-phrase-masked (security/mask-data "seed phrase")
         on-success         #(prn "success")
         on-error           #(prn "error")
-        expected           {:fx [[:multiaccount/validate-mnemonic
+        expected           {:fx [[:effects.profile/validate-recovery-phrase
                                   [seed-phrase-masked on-success on-error]]]}]
     (is (= expected
            (sut/wallet-validate-seed-phrase

--- a/src/status_im/contexts/wallet/add_account/create_account/events.cljs
+++ b/src/status_im/contexts/wallet/add_account/create_account/events.cljs
@@ -61,16 +61,6 @@
 
 (rf/reg-event-fx :wallet/seed-phrase-validated seed-phrase-validated)
 
-(defn seed-phrase-entered
-  [_ [seed-phrase on-error]]
-  {:fx [[:multiaccount/validate-mnemonic
-         [seed-phrase
-          (fn [mnemonic key-uid]
-            (rf/dispatch [:wallet/seed-phrase-validated mnemonic key-uid on-error]))
-          on-error]]]})
-
-(rf/reg-event-fx :wallet/seed-phrase-entered seed-phrase-entered)
-
 (defn store-account-generated-with-mnemonic
   [{:keys [db]} [{:keys [new-account-data keypair-name]}]]
   (let [new-account (update new-account-data :mnemonic security/mask-data)]

--- a/src/status_im/contexts/wallet/add_account/create_account/select_keypair/view.cljs
+++ b/src/status_im/contexts/wallet/add_account/create_account/select_keypair/view.cljs
@@ -16,13 +16,21 @@
    [[{:icon                :i/add
       :accessibility-label :generate-new-keypair
       :label               (i18n/label :t/generate-new-keypair)
-      :on-press            #(rf/dispatch [:navigate-to :screen/wallet.backup-recovery-phrase])}
+      :on-press            #(rf/dispatch [:navigate-to :screen/backup-recovery-phrase
+                                          {:on-success (fn [{:keys [masked-seed-phrase random-phrase]}]
+                                                         (rf/dispatch [:wallet/store-new-seed-phrase
+                                                                       {:seed-phrase masked-seed-phrase
+                                                                        :random-phrase
+                                                                        random-phrase}]))}])}
      {:icon                :i/seed
       :accessibility-label :import-using-phrase
       :label               (i18n/label :t/import-using-phrase)
       :add-divider?        true
-      :on-press            #(rf/dispatch [:navigate-to :screen/wallet.enter-seed-phrase
-                                          {:recovering-keypair? true}])}
+      :on-press            (fn []
+                             (rf/dispatch [:navigate-to :screen/use-recovery-phrase
+                                           {:on-success (fn [{:keys [key-uid phrase on-error]}]
+                                                          (rf/dispatch [:wallet/seed-phrase-validated
+                                                                        phrase key-uid on-error]))}]))}
      (when (ff/enabled? ::ff/wallet.import-private-key)
        {:icon                :i/key
         :accessibility-label :import-private-key

--- a/src/status_im/contexts/wallet/effects.cljs
+++ b/src/status_im/contexts/wallet/effects.cljs
@@ -4,6 +4,7 @@
     [native-module.core :as native-module]
     [promesa.core :as promesa]
     [status-im.common.json-rpc.events :as json-rpc]
+    [status-im.contexts.profile.recover.effects :as profile.recover.effects]
     [taoensso.timbre :as log]
     [utils.re-frame :as rf]
     [utils.security.core :as security]
@@ -25,26 +26,6 @@
       {:MnemonicPhrase phrase
        :paths          paths}
       on-success))))
-
-(defn validate-mnemonic
-  [mnemonic]
-  (-> mnemonic
-      (security/safe-unmask-data)
-      (native-module/validate-mnemonic)
-      (promesa/then (fn [result]
-                      (let [{:keys [keyUID]} (transforms/json->clj result)]
-                        {:key-uid keyUID})))))
-
-(rf/reg-fx
- :multiaccount/validate-mnemonic
- (fn [[mnemonic on-success on-error]]
-   (-> (validate-mnemonic mnemonic)
-       (promesa/then (fn [{:keys [key-uid]}]
-                       (when (fn? on-success)
-                         (on-success mnemonic key-uid))))
-       (promesa/catch (fn [error]
-                        (when (and error (fn? on-error))
-                          (on-error error)))))))
 
 (defn create-account-from-private-key
   [private-key]
@@ -81,7 +62,7 @@
 
 (defn import-missing-keypair-by-seed-phrase
   [keypair-key-uid seed-phrase password]
-  (-> (validate-mnemonic seed-phrase)
+  (-> (profile.recover.effects/validate-mnemonic seed-phrase)
       (promesa/then
        (fn [{:keys [key-uid]}]
          (if (not= keypair-key-uid key-uid)

--- a/src/status_im/navigation/screens.cljs
+++ b/src/status_im/navigation/screens.cljs
@@ -47,6 +47,7 @@
     [status-im.contexts.preview.quo.component-preview.view :as component-preview]
     [status-im.contexts.preview.quo.main :as quo.preview]
     [status-im.contexts.preview.status-im.main :as status-im-preview]
+    [status-im.contexts.profile.backup-recovery-phrase.view :as backup-recovery-phrase]
     [status-im.contexts.profile.contact.share.view :as share-contact]
     [status-im.contexts.profile.contact.view :as contact-profile]
     [status-im.contexts.profile.edit.accent-colour.view :as edit-accent-colour]
@@ -107,8 +108,6 @@
      wallet-import-private-key]
     [status-im.contexts.wallet.add-account.create-account.key-pair-name.view :as
      wallet-key-pair-name]
-    [status-im.contexts.wallet.add-account.create-account.new-keypair.backup-recovery-phrase.view :as
-     wallet-backup-recovery-phrase]
     [status-im.contexts.wallet.add-account.create-account.new-keypair.confirm-backup.view :as
      wallet-confirm-backup]
     [status-im.contexts.wallet.add-account.create-account.select-keypair.view :as wallet-select-keypair]
@@ -476,9 +475,9 @@
      :options   {:insets {:top? true}}
      :component wallet-create-account/view}
 
-    {:name      :screen/wallet.backup-recovery-phrase
+    {:name      :screen/backup-recovery-phrase
      :options   {:insets {:top? true :bottom? true}}
-     :component wallet-backup-recovery-phrase/view}
+     :component backup-recovery-phrase/view}
 
     {:name      :screen/wallet.confirm-backup
      :options   {:insets {:top? true :bottom? true}}
@@ -488,7 +487,7 @@
      :options   {:insets {:top? true}}
      :component wallet-key-pair-name/view}
 
-    {:name      :screen/wallet.enter-seed-phrase
+    {:name      :screen/use-recovery-phrase
      :component enter-seed-phrase/view}
 
     {:name      :screen/wallet.share-address

--- a/src/status_im/subs/profile.cljs
+++ b/src/status_im/subs/profile.cljs
@@ -345,6 +345,12 @@
  (fn [{:keys [processing]}]
    processing))
 
+(re-frame/reg-sub
+ :profile/recovery-phrase-backed-up?
+ :<- [:profile/profile]
+ (fn [profile]
+   (not (boolean (seq (:mnemonic profile))))))
+
 ;; LINK PREVIEW
 ;; ========================================================================================================
 

